core/geo/src/main/java/datawave/core/geo/function/AbstractGeoFunctionDetails.java

```diff
@@ -0,0 +1,132 @@
+package datawave.core.geo.function;
+
+import static datawave.core.query.jexl.nodes.QueryPropertyMarker.MarkerType.BOUNDED_RANGE;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.jexl3.parser.ASTGENode;
+import org.apache.commons.jexl3.parser.ASTLENode;
+import org.apache.commons.jexl3.parser.ASTTrueNode;
+import org.apache.commons.jexl3.parser.JexlNode;
+import org.apache.commons.jexl3.parser.ParserTreeConstants;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+
+import datawave.core.geo.utils.CommonGeoUtils;
+import datawave.core.geo.utils.GeoQueryConfig;
+import datawave.core.query.jexl.JexlNodeFactory;
+import datawave.core.query.jexl.nodes.QueryPropertyMarker;
+import datawave.data.type.GeoType;
+import datawave.data.type.GeometryType;
+import datawave.data.type.PointType;
+import datawave.data.type.Type;
+
+public abstract class AbstractGeoFunctionDetails {
+    public abstract String getNamespace();
+
+    public abstract String getName();
+
+    public abstract Set<String> getFields();
+
+    public abstract List<Geometry> getGeometry();
+
+    public abstract List<Envelope> getEnvelope();
+
+    public JexlNode generateIndexNode(Map<String,Set<Type<?>>> typesByField, GeoQueryConfig config) {
+        Map<String,List<String[]>> termsAndRangesByField = generateTermsAndRangesByField(typesByField, config);
+        Map<String,List<JexlNode>> indexNodesByField = generateIndexNodesByField(termsAndRangesByField);
+
+        List<JexlNode> orNodes = new ArrayList<>();
+        for (List<JexlNode> indexNodes : indexNodesByField.values()) {
+            if (!indexNodes.isEmpty()) {
+                orNodes.add(indexNodes.size() > 1 ? JexlNodeFactory.createOrNode(indexNodes) : indexNodes.get(0));
+            }
+        }
+
+        JexlNode finalNode;
+        if (!orNodes.isEmpty()) {
+            finalNode = orNodes.size() > 1 ? JexlNodeFactory.createOrNode(orNodes) : orNodes.get(0);
+        } else {
+            finalNode = new ASTTrueNode(ParserTreeConstants.JJTTRUENODE);
+        }
+
+        return finalNode;
```

Comment:
When a method ends on conditionally building a junction, I'm partial to the following syntax
```java
switch(orNodes.getNumChildren()){
  case 0:
    return new ASTTrueNode();
  case 1:
    return orNodes.get(0);
  default:
    return JexlNodeFactory.createOrNode(orNodes);
}
```

Reply:
i don't think what you have is any more/less readable than what i currently have.  the current logic is less lines of code though.

i don't like multiple return statements in a method either.